assets/js/base/components/text-input/style.scss

```diff
@@ -2,6 +2,18 @@
 	position: relative;
 	margin-top: $gap;
 	white-space: nowrap;
+
+	input {
+		border: 1px solid $input-border-gray;
+	}
+
+	&.has-error input {
+		border: 1px solid $error-red;
```

Comment:
This is needed because storefront sets the border to 0 and adds a shadow to inputs.  So I think this should be an explicit rule that overrides _any_ themes setting for input border (on errors).

Reply:
How about trying outline instead? Outlines don't affect layout.

Reply:
Outline looks bad (focus is lost on applying the coupon so it will be with non-rounded corners initially).

Also tried shadow as per Mike's suggestion but I think it looks bad too so just went with forcing the border using `!important`.  We can revisit styles in our styles review as necessary but this seems to be the better option here because I think we want to make sure this error condition is clear (and make it harder for themes to override).